PWGHF/TableProducer/trackIndexSkimCreator.cxx

```diff
@@ -530,33 +569,47 @@ struct HfTrackIndexSkimCreatorTagSelTracks {
       }
     }
 
-    iDebugCut = 5;
-    // DCA cut
-    if ((debug || statusProng > 0)) {
-      if ((debug || TESTBIT(statusProng, CandidateType::Cand2Prong)) && !isSelectedTrackDCA(trackPt, dca, CandidateType::Cand2Prong)) {
-        CLRBIT(statusProng, CandidateType::Cand2Prong);
-        if (debug) {
-          // cutStatus[CandidateType::Cand2Prong][3] = false;
-          if (fillHistograms) {
-            registry.fill(HIST("hRejTracks"), (nCuts + 1) * CandidateType::Cand2Prong + iDebugCut);
-          }
+    // quality cut for soft pion
+    hasGoodQuality = true;
+    if (doCutQuality.value && (debug || TESTBIT(statusProng, CandidateType::CandDstar))) {
+      if (useIsGlobalTrackForSoftPion) {
+        if (hfTrack.isGlobalTrack() != (uint8_t) true) {
+          hasGoodQuality = false;
+        }
+      } else if (useIsGlobalTrackWoDCAForSoftPion) {
+        if (hfTrack.isGlobalTrackWoDCA() != (uint8_t) true) {
+          hasGoodQuality = false;
+        }
+      } else if (useIsQualityTrackITSForSoftPion) {
+        if (hfTrack.isQualityTrackITS() != (uint8_t) true) {
+          hasGoodQuality = false;
+        }
+      } else {
+        UChar_t clustermap = hfTrack.itsClusterMap();
+        if ((hfTrack.flags() & o2::aod::track::ITSrefit && (TESTBIT(clustermap, 0) || TESTBIT(clustermap, 1)))) {
+          hasGoodQuality = false;
```

Comment:
Indeed it was the opposite, thanks for spotting! I fixed it. However this was originally meant for the analysis of converted Run2 data, since the Run3 selections should be applied using the track-selection flags (that's why it only checks the first two layers). Here I used the same logic as for the other tracks (you can see the previous `if` block).

Reply:
Add a comment in the code specifying that it is meant for Run 2 converted data

Reply:
Done